File: leetcode/C++/the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        vector<pair<int,int>> v;
        vector<int> sol;
        for(int i=0, num=0; i<mat.size(); ++i){
            num = 0;
            for(int j=0; j<mat[0].size(); ++j) if(mat[i][j]==1) ++num;
            v.push_back({num, i});
        }
        sort(v.begin(), v.end());
        for(int i=0; i<k; ++i) sol.push_back(v[i].second);
        return sol;
    }
};

File: leetcode/Python/the-k-weakest-rows-in-a-matrix.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        v, sol, val = list(), list(), 0
        for i in range(len(mat)):
            val = 0
            for j in range(len(mat[0])):
                if mat[i][j] == 1:
                    val += 1
            v.append([val, i])
        v.sort()
        for i in range(k):
            sol.append(v[i][1])
        return sol